File: backend/src/main/kotlin/ru/itmo/db/coursework/api/mapping/TeaDtoMapper.kt
package ru.itmo.db.coursework.api.mapping

import org.springframework.stereotype.Service
import ru.itmo.db.coursework.api.dto.TeaDto
import ru.itmo.db.coursework.model.Tea

@Service
class TeaDtoMapper {
    fun fromDto(dto: TeaDto) = with(dto, {
        Tea(
                id = id,
                name = name,
                price = price,
                description = description
        )
    })

    fun toDto(model: Tea) = with(model, {
        TeaDto(
                id = id,
                name = name,
                price = price,
                description = description
        )
    })
}

File: frontend/src/api/mapping/TeaDtoMapper.ts
import TeaDto from "$src/api/dto/TeaDto";
import Tea from "$src/game/model/Tea";

export default class TeaDtoMapper {
  public static fromDto(dto: TeaDto) {
    return new Tea(dto.id, dto.name, dto.price, dto.description)
  }

  public static toDto(model: Tea) {
    return new TeaDto(model.id, model.name, model.price, model.description)
  }
}